database/factories/BeatmapsetNominationFactory.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+namespace Database\Factories;
+
+use App\Models\BeatmapsetEvent;
+use App\Models\BeatmapsetNomination;
+use Illuminate\Database\Eloquent\Factories\Factory;
+
+class BeatmapsetNominationFactory extends Factory
+{
+    /**
+     * The name of the factory's corresponding model.
+     *
+     * @var string
+     */
+    protected $model = BeatmapsetNomination::class;
+
+    public function configure()
+    {
+        return $this->afterCreating(function (BeatmapsetNomination $nomination) {
+            $event = $nomination->beatmapset->events()->create([
```

Comment:
maybe check whether or not `$nomination->event_id` is null before creating a new one? :thinking:

Reply:
the factory one only exists to be overwritten by the one from the event, though 🤔